airflow/jobs/backfill_job.py

```diff
@@ -218,6 +218,13 @@ def _update_counters(self, ti_status, session=None):
                 ti_status.running.pop(reduced_key)
                 ti_status.to_run[ti.key] = ti
 
+            elif ti.state == TaskInstanceState.SCHEDULED:
+                # Deferrable task can go from DEFERRED to SCHEDULED,
+                # when that happens, we need to remove it from running
+                # and add it to to_run
+                ti_status.running.pop(ti.key)
+                ti_status.to_run[ti.key] = ti
```

Comment:
```suggestion
            # special case: Deferrable task can go from DEFERRED to SCHEDULED;
            # when that happens, we need to put it back as in UP_FOR_RESCHEDULE
            elif ti.state == TaskInstanceState.SCHEDULED:
                self.log.debug("Task instance %s is resumed from deferred state", ti)
                ti_status.running.pop(ti.key)
                ti_status.to_run[ti.key] = ti
```

Match styles of other blocks. Perhaps put this block right after the `UP_FOR_RESCHEDULE` one for clarity?

Reply:
As regards the comment, it turns out that kubernetes can also clear queued runs and they'd get to the scheduled state. This can get the tasks stuck in scheduled state. While this fix will fix that by having the backfill run it again, I'm torn between modifying the comment +log here and not clearing backfill runs in kubernetes. WDYT? 
If we don't allow kubernetes to clear the queued runs that have overstayed, we may run into a case where a backfill task would be stuck in queued.

Reply:
It does not feel right to special case for Kubernetes here though 🤔